### ISSUE TITLE
Build fails when using `#include <>` and script_phase

### ISSUE THREAD

**Issue by nick-airturn** [CocoaPods/CocoaPods#8201]
* [ ] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

In podspec, use s.script_phase to add a script phase
pod uses a compiled dynamic framework
In code, use `#import <Framework/Framework.h>`
Run pod install and then build with Xcode

## What did you expect to happen?

App would build successfully.

## What happened instead?

Without using script_phases, build succeeds. Using script_phases, build fails.
When building with script_phases the compile command uses this framework flag:
`-F/Users/nick/Library/Developer/Xcode/DerivedData/Cocoapods_Test-dzgqqjesmefrglhipacuztdtvcvj/Build/Products/Debug-iphonesimulator/<framework name>`
This directory does not contain the complete framework, hence the compilation fails.
When building without script_phases it uses this framework flag:
`-F<path to project root>/Pods/<path to framework parent directory from inside zip>`

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14 (18A391)
       Xcode : 10.0 (10A255)
         Git : git version 2.19.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 71c5aa7b4563459a4235f20e495feb1a9d5f356e
```

### Installation Source

```
Executable Path: /usr/local/Cellar/cocoapods/1.5.3/libexec/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

This framework is not public, however I believe it would work with any project using an existing compiled framework.

**Comment by nick-airturn** [comment]
I think this is failing because cocoapods is adding a target for the framework where it compiles a 'dummy' framework `Compile <framework name>-dummy.c` in which to run the build script. The problem is it then uses this as the framework.

The build script phase I want to add strips the framework of binaries for other platforms ready for archiving.

**Comment by amorde** [comment]
@nick-airturn can you explain what you mean when you say "with script phases" and "without script phases"? is this using the script phases DSL in the Podfile / Podspec?

A sample project would be helpful here

**Comment by segiddins** [comment]
I think this might be fixed in the 1.6 beta?

**Comment by nrbrook** [comment]
Yes it seems to be fixed in 1.6 beta 2, thanks!

I saw it mentioned somewhere that a binary stripping script is not required as cocoapods does it for you (for submitting apps to the App store with only device architectures) but I can't find any docs to confirm that, is that true or do I still need a script?

**Comment by amorde** [comment]
> I saw it mentioned somewhere that a binary stripping script is not required as cocoapods does it for you (for submitting apps to the App store with only device architectures) but I can't find any docs to confirm that, is that true or do I still need a script?

this is true - you can check out the code that handles the stripping part of the script (at the time this comment was written) here: 

https://github.com/CocoaPods/CocoaPods/blob/2980c4cb14860525c60fc2ef7e92ee8b4c4d66c8/lib/cocoapods/generator/embed_frameworks_script.rb#L172-L197
